packages/ag-charts-community/src/chart/series/polar/pieSeries.ts

```diff
@@ -1823,7 +1826,9 @@ export class PieSeries extends PolarSeries<PieNodeDatum> {
                     sector.fillOpacity = fillOpacity;
                 },
                 onComplete: () => {
-                    if (cleanup) this.resetSectors();
+                    // TODO: Calling reset here causes a discrepancy between the data and the sectors, such that
+                    // sectors that were previously removed then shown do not get rendered.
+                    // if (cleanup) this.resetSectors();
```

Comment:
Theoretically, this should be fine. I've put these "reset" functions at the end of animations mostly to cover my back in case the animation goes wrong. However in this situation, it was causing the issue. It's something I should delve deeper into at some point, but this fixes it for now.